nmrfx-analyst-gui/src/main/java/org/nmrfx/analyst/gui/molecule/CanvasMolecule.java

```diff
@@ -333,52 +325,83 @@ String getHit() {
+        if (!molecule.globalSelected.isEmpty()) {
+            molecule.clearSelected();
+        }
+        if (selectMode && selectable) {
+            selected = true;
+        }
+    }
+
+
     @Override
-    public boolean hit(double x, double y) {
+    public boolean hit(double x, double y, boolean selectMode) {
         if (!bounds2D.contains(x, y)) {
             hitAtom = -1;
+            if (selectMode) {
+                selected = false;
+            }
             return false;
         } else {
             hitAtom = pick(null, x, y);
             startX1 = bx1;
             startY1 = by1;
             startX2 = bx2;
             startY2 = by2;
-
             if (hitAtom == -1) {
-                Molecule molecule = Molecule.get(molName);
-                if (!molecule.globalSelected.isEmpty()) {
-                    molecule.clearSelected();
-                }
-                return true;
+                selectMolecule(selectMode);
             } else {
-                System.out.println(getHit());
-                String aName = getHit();
-                Molecule molecule = Molecule.get(molName);
-                try {
-                    molecule.selectAtoms(aName);
-                } catch (InvalidMoleculeException ex) {
-                    log.warn(ex.getMessage(), ex);
-                }
-
-                return true;
+                selectAtom(selectMode);
             }
+            return true;
         }
     }
 
     @Override
     public void move(double[][] bounds, double[][] world, double[] start, double[] pos) {
         double dx = pos[0] - start[0];
         double dy = pos[1] - start[1];
-        bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
-        bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
-        by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
-        by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+        if (activeHandle < 0) {
+            bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
+            bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
+            by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
+            by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+
+        }
+        if (activeHandle == 0) {
+            bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
+            by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
+        }
+        if (activeHandle == 1) {
+            bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
+            by1 = yPosType.move(startY1, dy, bounds[1], world[1]);
+        }
+        if (activeHandle == 2) {
+            bx2 = xPosType.move(startX2, dx, bounds[0], world[0]);
+            by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+        }
+        if (activeHandle == 3) {
+            bx1 = xPosType.move(startX1, dx, bounds[0], world[0]);
+            by2 = yPosType.move(startY2, dy, bounds[1], world[1]);
+        }
```

Comment:
I think this should really be a series of it..else if, otherwise the condition gets checked for all even if the first condition passes